components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -1124,4 +1124,21 @@ public static boolean isUseVerifyClaimEnabled() {
         return Boolean.parseBoolean(IdentityUtil.getProperty
                 (IdentityRecoveryConstants.ConnectorConfig.USE_VERIFY_CLAIM_ON_UPDATE));
     }
+
+    /**
+     * Clones the given map.
+     *
+     * @param map          Map.
+     * @return             Cloned Map.
+     */
+    public static Map<String, Object> cloneMap(Map<String, Object> map) {
+        if (map == null) {
```

Comment:
Shall we add a new line after the method signature? Can't we use `MapUtils.isEmpty()`?

Reply:
Why do we need to deep-copy the map in places where this is used?

Reply:
-1 to add this as a Utils method since this is not related to Identity Recovery. Better if we can add it as a private method in the classes that use this IMO.

Reply:
Fixed in 61d7548fc71aca19ecb53cb10928da2bb00294d6